tests/python/test_dynamic.py

```diff
@@ -265,3 +265,28 @@ def make_list():
 
     for i in range(20):
         assert x[i] == i * i * i * 10000000000
+
+
+@test_utils.test(require=ti.extension.sparse, exclude=[ti.metal])
```

Comment:
Comment on why `ti.metal` gets excluded?

Reply:
Because according to @k-ye we may give up sparse support for metal backend. So for now I'm not considering metal when implementing dynamic SNode (Since Metal backend does not support dynamic memory allocating, we just allocate the whole memory when we materialize it. So it's easier than LLVM-based backends to implement if we decide not to give up metal backend).